lms/djangoapps/django_comment_client/base/tests.py

```diff
@@ -1906,3 +1921,267 @@ def test_requires_matched_user_has_forum_content(self, mock_request):
+                org='TestX',
+                course='TR-101',
+                run='Event_Transform_Test',
+                default_store=ModuleStoreEnum.Type.mongo,
+            ),
+            ModuleStoreEnum.Type.split: CourseFactory.create(
+                org='TestX',
+                course='TR-101S',
+                run='Event_Transform_Test_Split',
+                default_store=ModuleStoreEnum.Type.split,
+            ),
+        }
+        self.course = self.courses_by_store['mongo']
+        self.student = UserFactory.create()
+        self.staff = UserFactory.create(is_staff=True)
+        UserBasedRole(user=self.staff, role=CourseStaffRole.ROLE).add_course(self.course.id)
+        CourseEnrollmentFactory.create(user=self.student, course_id=self.course.id)
+        self.category = ItemFactory.create(
+            parent_location=self.course.location,
+            category='discussion',
+            discussion_id=self.CATEGORY_ID,
+            discussion_category=self.PARENT_CATEGORY_NAME,
+            discussion_target=self.CATEGORY_NAME,
+        )
+        self.team_category = ItemFactory.create(
+            parent_location=self.course.location,
+            category='discussion',
+            discussion_id=self.TEAM_CATEGORY_ID,
+            discussion_category=self.TEAM_PARENT_CATEGORY_NAME,
+            discussion_target=self.TEAM_CATEGORY_NAME,
+        )
+        CourseTeamFactory(
+            name='Team 1',
+            course_id=self.course.id,
+            topic_id='arbitrary-topic-id',
+            discussion_topic_id=self.TEAM_CATEGORY_ID,
+        )
+
+    def test_empty_event(self):
+        self.assertDictEqual(_get_transformed_event({}), {'event': {}})
+
+    def test_non_mobile_source(self):
+        event, event_trans = _create_and_transform_event(
+            event_source='server',
+            course_id=self.course.id,
+            discussion_id=self.DUMMY_COMMENTABLE_ID,
+            thread_id=self.DUMMY_THREAD_ID,
+        )
+        self.assertDictEqual(event_trans, event)
+
+    def test_bad_field_types(self):
+        # Just make sure no exception is raised
+        _, event_trans = _create_and_transform_event(
+            event_source=[],
+            context='abc',
+            course_id={},
+            discussion_id=3,
+            thread_id=object(),
+            title=3.14,
+        )
+
+    def test_bad_course_id(self):
+        # Just make sure no exception is raised
+        _create_and_transform_event(course_id='non-existent-course-id')
+
+    def test_bad_username(self):
+        # Just make sure no exception is raised
+        _create_and_transform_event(username='non-existent-username')
+
+    def test_bad_url(self):
+        # Just make sure no exception is raised
+        _create_and_transform_event(
+            course_id=self.course.id,
+            commentable_id='malformed/commentable/id',
+            thread_id='malformed/thread/id',
+        )
+
```

Comment:
@thallada Reposting your comments here (which became outdated when I moved this file):

In these four "bad" input cases, do we expect the input event to equal the transformed event because the transformer can't do anything with the bad data? It probably would be good to add in a `self.assertDictEqual(event_trans, event)` to each of them to ensure that.